px/px_process.py

```diff
@@ -222,46 +222,44 @@ def resolve_links(processes, now):
     Also, all processes will have a (possibly empty) "children" field containing
     a set of references to child processes.
     """
-    pid2process = {}
-    for process in processes:
-        # Guard against duplicate PIDs
-        assert process.pid not in pid2process
-
-        pid2process[process.pid] = process
-
+    for process in processes.values():
         process.children = set()
 
-    if 0 not in pid2process:
+    if 0 not in processes:
         kernel_process = create_kernel_process(now)
+        processes[0] = kernel_process
 
-        processes.append(kernel_process)
-        pid2process[0] = kernel_process
-
-    for process in processes:
+    for process in processes.values():
         if process.pid == 0:
             process.parent = None
         else:
-            process.parent = pid2process[process.ppid]
+            process.parent = processes[process.ppid]
             process.parent.children.add(process)
 
 
+def remove_process_and_descendants(processes, pid):
+    process = processes[pid]
+    process.parent.children.remove(process)
```

Comment:
Do we need to check that the process parent isn't `None`?

But since only PID 0 has no parent and we're unlikely to want to remove that I think this is fine as it is.

Yay for CI, without that I'd just have guessed this fails all the time :).